db/migrate/20231019052851_add_devise_to_users.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.0]
+  def self.up
+    change_table :users do |t|
+       ## Database authenticatable
+       t.string :email,              null: false, default: ""
+       t.string :encrypted_password, null: false, default: ""
+
+       ## Recoverable
+       t.string   :reset_password_token
+       t.datetime :reset_password_sent_at
+
+       ## Rememberable
+       t.datetime :remember_created_at
+
+       ## Trackable
+       # t.integer  :sign_in_count, default: 0, null: false
+       # t.datetime :current_sign_in_at
+       # t.datetime :last_sign_in_at
+       # t.string   :current_sign_in_ip
+       # t.string   :last_sign_in_ip
+
+       t.string   :confirmation_token
+       t.datetime :confirmed_at
+       t.datetime :confirmation_sent_at
+       t.string   :unconfirmed_email # Only if using reconfirmable
+
+       ## Lockable
+       # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+       # t.string   :unlock_token # Only if unlock strategy is :email or :both
+       # t.datetime :locked_at
+
+
+       # Uncomment below if timestamps were not included in your original model.
+       # t.timestamps null: false
+     end
+
+     add_index :users, :email
+     add_index :users, :reset_password_token, unique: true
+     # add_index :users, :confirmation_token,   unique: true
+     # add_index :users, :unlock_token,         unique: true
+   end
+
+   def self.down
+     # By default, we don't want to make any assumption about how to roll back a migration when your
+     # model already existed. Please edit below which fields you would like to remove in this migration.
+     raise ActiveRecord::IrreversibleMigration
+   end
+ end
```

Comment:
- [ ] Although I am highlighting this migration file, the suggested change should be made to the following migration file;
`20231003124753_change_counter_columns.rb`

---
When I try to run all the migration files to create all the necessary tables and columns in the database, I get this error:

![image](https://github.com/kkmanuu/Blog-app/assets/77106962/17972593-e1ac-4b2a-bd4b-7350cf017010)

---
In the migration file for creating the `Users` table, the correct column name is `post_counter` and not `posts_counter`. The same applies to the `comment_counter` and `like_counter` in the migration file for creating the `Posts` table. 

Kindly change the names of these columns in the `20231003124753_change_counter_columns.rb` migration file. 👍 🖱️

---

#### Hint: If you want to rename the column, the correct way to do it is by:

```ruby
  def change
    rename_column :users, old_column_name, new_column_name
  end

``` 

---

#### N.B: Kindly change all references to old column names to the new ones in different files where they are used. 😎  💻

Reply:
@Gambit142 Done making the changes